File: pose_analysis_app/tools/generate_icon.dart
import 'dart:io';
import 'dart:math';
import 'dart:typed_data';
import 'dart:ui' as ui;

import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

class IconGenerator {
  static Future<void> generate() async {
    final recorder = ui.PictureRecorder();
    final canvas = Canvas(recorder, const Rect.fromLTWH(0, 0, 1024, 1024));

    final bgPaint = Paint()
      ..shader = ui.Gradient.linear(
        const Offset(0, 0),
        const Offset(1024, 1024),
        [const Color(0xFF6C63FF), const Color(0xFF00D4AA)],
      );
    canvas.drawRRect(
      RRect.fromRectAndRadius(
        const Rect.fromLTWH(0, 0, 1024, 1024),
        const Radius.circular(200),
      ),
      bgPaint,
    );

    final circlePaint = Paint()..color = Colors.white.withOpacity(0.12);
    canvas.drawCircle(const Offset(512, 440), 300, circlePaint);

    final bodyPaint = Paint()
      ..color = Colors.white
      ..strokeWidth = 20
      ..strokeCap = StrokeCap.round
      ..style = PaintingStyle.stroke;

    final dotPaint = Paint()
      ..color = Colors.white
      ..style = PaintingStyle.fill;

    final head = Offset(512, 260);
    final neck = Offset(512, 330);
    final lShoulder = Offset(410, 380);
    final rShoulder = Offset(614, 380);
    final lElbow = Offset(330, 490);
    final rElbow = Offset(694, 490);
    final lWrist = Offset(360, 580);
    final rWrist = Offset(664, 580);
    final hip = Offset(512, 530);
    final lKnee = Offset(430, 670);
    final rKnee = Offset(594, 670);
    final lAnkle = Offset(410, 800);
    final rAnkle = Offset(614, 800);

    final connections = [
      [head, neck], [neck, lShoulder], [neck, rShoulder],
      [lShoulder, lElbow], [rShoulder, rElbow],
      [lElbow, lWrist], [rElbow, rWrist],
      [neck, hip], [hip, lKnee], [hip, rKnee],
      [lKnee, lAnkle], [rKnee, rAnkle],
    ];

    for (final c in connections) {
      canvas.drawLine(c[0], c[1], bodyPaint);
    }

    final points = [
      head, neck, lShoulder, rShoulder, lElbow, rElbow,
      lWrist, rWrist, hip, lKnee, rKnee, lAnkle, rAnkle,
    ];
    for (final p in points) {
      canvas.drawCircle(p, 14, dotPaint);
      canvas.drawCircle(p, 14, Paint()
        ..color = const Color(0xFF6C63FF)
        ..style = PaintingStyle.stroke
        ..strokeWidth = 4);
    }

    final picture = recorder.endRecording();
    final image = await picture.toImage(1024, 1024);
    final byteData = await image.toByteData(format: ui.ImageByteFormat.png);
    final bytes = byteData!.buffer.asUint8List();

    final iconFile = File('assets/icon/app_icon.png');
    await iconFile.create(recursive: true);
    await iconFile.writeAsBytes(bytes);

    final sizes = {
      'mipmap-mdpi': 48,
      'mipmap-hdpi': 72,
      'mipmap-xhdpi': 96,
      'mipmap-xxhdpi': 144,
      'mipmap-xxxhdpi': 192,
    };

    for (final entry in sizes.entries) {
      final dir = Directory('android/app/src/main/res/${entry.key}');
      if (dir.existsSync()) {
        final resized = await _resize(bytes, entry.value);
        await File('${dir.path}/ic_launcher.png').writeAsBytes(resized);
        await File('${dir.path}/ic_launcher_round.png').writeAsBytes(resized);
      }
    }

    print('App icon generated successfully!');
    image.dispose();
  }

  static Future<Uint8List> _resize(Uint8List src, int size) async {
    final codec = await ui.instantiateImageCodec(src);
    final frame = await codec.getNextFrame();
    final img = frame.image;

    final r = ui.PictureRecorder();
    final c = Canvas(r, Rect.fromLTWH(0, 0, size.toDouble(), size.toDouble()));
    c.drawImageRect(
      img,
      Rect.fromLTWH(0, 0, img.width.toDouble(), img.height.toDouble()),
      Rect.fromLTWH(0, 0, size.toDouble(), size.toDouble()),
      Paint()..filterQuality = FilterQuality.high,
    );

    final pic = r.endRecording();
    final out = await pic.toImage(size, size);
    final bd = await out.toByteData(format: ui.ImageByteFormat.png);
    img.dispose();
    out.dispose();
    return bd!.buffer.asUint8List();
  }
}

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await IconGenerator.generate();
  // ignore: avoid_print
}

File: pose_analysis_app/tools/generate_icon.py
from PIL import Image, ImageDraw, ImageFont
import os

def create_icon():
    size = 1024
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    cx, cy = size // 2, size // 2
    r = size // 2

    for y in range(size):
        for x in range(size):
            dx = x - cx
            dy = y - cy
            dist = (dx*dx + dy*dy) ** 0.5
            if dist <= r:
                t = y / size
                c1 = (108, 63, 255)
                c2 = (0, 150, 255)
                red = int(c1[0] * (1 - t) + c2[0] * t)
                green = int(c1[1] * (1 - t) + c2[1] * t)
                blue = int(c1[2] * (1 - t) + c2[2] * t)
                draw.point((x, y), fill=(red, green, blue, 255))

    white = (255, 255, 255, 255)
    bar_w = 280
    bar_h = 56
    bx = cx - bar_w // 2
    by = cy - bar_h // 2

    draw.rounded_rectangle(
        [bx, by, bx + bar_w, by + bar_h],
        radius=bar_h // 2,
        fill=white,
    )

    plate_configs = [
        (70, 220, 0),
        (55, 180, 0),
        (42, 140, 0),
    ]

    for side in [-1, 1]:
        base_x = cx + side * (bar_w // 2 + 10)
        for i, (pw, ph, _) in enumerate(plate_configs):
            px = base_x + side * (sum(p[0] for p in plate_configs[:i]) + i * 8) * side
            if side == 1:
                px = base_x + sum(p[0] for p in plate_configs[:i]) + i * 8
            else:
                px = base_x - sum(p[0] for p in plate_configs[:i]) - i * 8 - pw

            py = cy - ph // 2
            draw.rounded_rectangle(
                [px, py, px + pw, py + ph],
                radius=8,
                fill=white,
            )

    grip_lines = [-50, -25, 0, 25, 50]
    for gx in grip_lines:
        draw.line(
            [cx + gx, by + 8, cx + gx, by + bar_h - 8],
            fill=(200, 200, 255, 120),
            width=4,
        )

    output = 'assets/icon/app_icon.png'
    os.makedirs(os.path.dirname(output), exist_ok=True)
    img.save(output, 'PNG')

    res_dir = 'android/app/src/main/res'
    sizes = {
        'mipmap-mdpi': 48,
        'mipmap-hdpi': 72,
        'mipmap-xhdpi': 96,
        'mipmap-xxhdpi': 144,
        'mipmap-xxxhdpi': 192,
    }

    for folder, sz in sizes.items():
        path = os.path.join(res_dir, folder)
        if os.path.exists(path):
            resized = img.resize((sz, sz), Image.LANCZOS)
            resized.save(os.path.join(path, 'ic_launcher.png'), 'PNG')
            resized.save(os.path.join(path, 'ic_launcher_round.png'), 'PNG')

    print(f'Icon generated: {output} (1024x1024)')

if __name__ == '__main__':
    create_icon()
